dashboardsSanitizer/src/main/java/org/opensearch/migrations/dashboards/model/Dashboard.java

```diff
@@ -0,0 +1,272 @@
+    private String created_at;
+
+    @JsonProperty("version")
+    private String version;
+
+    @JsonProperty("sort")
+    private long[] sort;
+
+    @JsonProperty("typeMigrationVersion")
+    private String typeMigrationVersion;
+
+    @JsonProperty("managed")
+    private boolean managed;
+
+    /**
+     * Attributes of the Dashboard object.
+     * @see Attributes
+     */
+    @Data
+    public static class Attributes {
+        @JsonProperty("description")
+        private String description;
+
+        @JsonProperty("hits")
+        private int hits;
+
+        @JsonProperty("kibanaSavedObjectMeta")
+        private SavedObjectMeta kibanaSavedObjectMeta;
+
+        @JsonProperty("optionsJSON")
+        private String optionsJSON;
+
+        @JsonProperty("panelsJSON")
+        private String panelsJSON;
+
+        @JsonProperty("locatorJSON")
+        private String locatorJSON;
+
+        @JsonProperty("timeRestore")
+        private boolean timeRestore;
+
+        @JsonProperty("title")
+        private String title;
+
+        // Index-pattern related START
+        @JsonProperty("runtimeFieldMap")
+        private String runtimeFieldMap;
+
+        @JsonProperty("fieldAttrs")
+        private String fieldAttrs;
+
+        @JsonProperty("fieldFormatMap")
+        private String fieldFormatMap;
+
+        @JsonProperty("fields")
+        private String fields;
+
+        @JsonProperty("sourceFilters")
+        private String sourceFilters;
+
+        @JsonProperty("timeFieldName")
+        private String timeFieldName;
+
+        @JsonProperty("typeMeta")
+        private String typeMeta;
+        //Index-patter related END
+
+        @JsonProperty("visualizationType")
+        private String visualizationType;
+
+        @JsonProperty("uiStateJSON")
+        private String uiStateJSON;
+
+        @JsonProperty("visState")
+        private String visState;
+
+        @JsonProperty("version")
+        private int version;
+
+        @JsonProperty("url")
+        private String url;
+
+        @JsonProperty("accessCount")
+        private int accessCount;
+
+        @JsonProperty("accessDate")
+        private long accessDate;
+
+        @JsonProperty("createDate")
+        private long createDate;
+
+        @JsonProperty("columns")
+        private String[] columns;
+
+        @JsonProperty("slug")
+        private String slug;
+
+        @JsonProperty("sort")
+        private String[][] sort;
+
+    }
+
+    /**
+     * This method makes the dashboard compatible with the OpenSearch,
+     * It sanitizes the panel JSON, removes incompatible object types, and updates the migration version accordingly.
+     * This is tested with kibana v 7.17, and not fully tested with Kibana v8 onwards
+     *
+     * @throws JsonProcessingException
+     * @throws IOException
+     */
+    public void makeCompatibleToOS() throws JsonProcessingException, IOException {
```

Comment:
Lets move this into the transformation project, in that project can you:
- You'll need to add a new entity called Document.
- Create a new transformation rule called something like `OSComptaibleDashboardObject`
- Make that class implementing `TransformationRule<Document>` with the functionality you've got in Dashboard.java

Please feel free to take inspiration from the existing transformation
`transformation/src/main/java/org/opensearch/migrations/transformation/rules/IndexMappingTypeRemoval.java`